### PR TITLE
e2e: add test cases for idempotence of initcontainers

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
#1222 was solved by #1224, but it is good to have test cases that check the idempotence of initcontainers

Signed-off-by: Hyeongju Johannes Lee <hyeongju.lee@intel.com>

### CODE REVIEW

**Comment by hj-johannes-lee** [comment]
The test for qat4 should fail before #1224 is merged, since the addition of these tests is for checking if initcontainers can run multiple times or not (and currently qat4 cannot).

But, dlb will be succeeded because the script already considered about that matter.

**Comment by bart0sh** [comment]
@hj-johannes-lee do we have only QAT and DLB initcontainers or other initcontainers are not idempotent?

**Comment by hj-johannes-lee** [comment]
@bart0sh I added dsa and iaa.!

File: test/e2e/utils/utils.go [hunk 1]
```diff
@@ -211,6 +211,23 @@ func TestPodsFileSystemInfo(pods []v1.Pod) error {
 	return nil
 }
 
+func TestInitcontainersRunIdempotent(f *framework.Framework, podList *v1.PodList, pluginName string) error {
```
Comment (mythi) [inline test/e2e/utils/utils.go:214]: this made me think what we discussed on using the operator in `test/e2e/<device>/*`: when using the operator this whole create / delete / re-create process is much simpler. We can just create and update (the same way how it's done in `test/envtest/*`.

Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:214]: Does the 'update' of the operator re run the initcontainer also?

Comment (mythi) [reply test/e2e/utils/utils.go:214]: `xDevicePlugin`  update triggers an update to the daemonSet. I believe the daemonSet controller then restarts the pods

**Comment by bart0sh** [comment]
> @bart0sh I added dsa and iaa.!

great! what about the rest?

**Comment by hj-johannes-lee** [comment]
> > @bart0sh I added dsa and iaa.!
> 
> great! what about the rest?

Umm, what else do we need to put?

When I asked Mikko since I was not sure,
he said `qat, qat4, iaa, dsa, dlb`...

Is there any other initcontainers that set devices?

**Comment by mythi** [comment]
> When I asked Mikko since I was not sure,
> he said `qat, qat4, iaa, dsa, dlb`...

I thought about it after Ed's question again. While we don't have initcontainers that do provisioning it might still have the same test case for all of the plugins to make sure a reinstall works.

**Comment by hj-johannes-lee** [comment]
> 

Aha, I see. It makes sense.! :) Let me add others.!

But, if we are gonna move to operator-based... then, do I just wait the operator-based pr to get merged and then change this to operator-based?

**Comment by bart0sh** [comment]
@hj-johannes-lee no changes in this PR for 12 days. What are we waiting for?

**Comment by hj-johannes-lee** [comment]
@bart0sh Waiting for Oleg's pr to be merged so that I can re-work with operator-based.

**Comment by bart0sh** [comment]
@hj-johannes-lee ok, thanks for the explanations. makes sense to me.

**Comment by bart0sh** [comment]
@hj-johannes-lee are we still waiting for something?

File: test/e2e/utils/utils.go (hunk 1 above)
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:215]: > xDevicePlugin update triggers an update to the daemonSet. I believe the daemonSet controller then restarts the pods

Do I apply this idea to dsa/iaa?
But could you give some info about this?
Idk how to update daemonset in e2e test.

Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:215]: Would you mean 'kubectl replace'?
I have no idea how to do the same as envtest..

**Comment by hj-johannes-lee** [comment]
@bart0sh I think I can continue.! Sorry, I was occupied by other many things..

**Comment by mythi** [comment]
It still fills an important gap. Perhaps submit an issue so that it's not forgotten?

**Pull request closed without merging**